airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py

```diff
@@ -50,31 +99,35 @@ def convert_from_sequential_state(
         ...
 
     @abstractmethod
-    def convert_to_sequential_state(self, cursor_field: "CursorField", stream_state: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
-        """
-        Convert the state message from the concurrency-compatible format to the stream's original format.
-
-        e.g.
-        { "created": 1617030403 }
-        """
-        ...
-
-    @abstractmethod
-    def increment(self, timestamp: Any) -> Any:
+    def increment(self, value: Any) -> Any:
         """
         Increment a timestamp by a single unit.
         """
         ...
 
-    @abstractmethod
     def merge_intervals(self, intervals: List[MutableMapping[str, Any]]) -> List[MutableMapping[str, Any]]:
         """
         Compute and return a list of merged intervals.
 
         Intervals may be merged if the start time of the second interval is 1 unit or less (as defined by the
         `increment` method) than the end time of the first interval.
         """
-        ...
+        if not intervals:
```

Comment:
This was moved from `DateTimeStreamStateConverter` as there is nothing specific to datetime and I expect all cursors to have the same implementation